internal/tablemap_test.go

```diff
@@ -91,3 +91,34 @@ func TestGetSpannerCol(t *testing.T) {
 		assert.Equal(t, spCol, spCol2, tc.name)
 	}
 }
+
+func TestGetSpannerKeyName(t *testing.T) {
+	conv := MakeConv()
+	schemaForeignKeys := make(map[string]bool)
+
+	basicTests := []struct {
+		name       string // Name of test.
+		srcKeyName string // Source foreign key name.
+		error      bool   // Whether an error is expected.
+		spKeyName  string // Expected Spanner foreign key name.
+	}{
+		{"Good name", "fktest", false, "fktest"},
+		{"Empty name", "", false, ""},
+		{"Collision", "fktest", false, "fktest_1"},
```

Comment:
I think this test case is a little confusing.

As noted in the comment on GetSpannerKeyName, each foreign key constraint name we encounter is unique. That's why we don't need to do consistent mapping of foreign key names -- the functionality isn't needed.

If this wasn't the case (and we did see the same foreign key constraint name during processing), then we'd probably have to do a consistent mapping.

So this test case is kind of odd/confusing: it tests a case that we don't need (and should never happen). And worse, it fixes our implementation -- if we even did need to implement consistent mapping, we'd have to fix these test cases. Or to put it another way: this testcase is testing a property of the implementation, not of the API.

Suggest deleting it (and the other similar tests).

Reply:
So i noticed that in postgres the constraint name can be duplicate in the schema if they are defined for different tables, but spanner doesn't accept that. 
So, this collision handling is specifically for postgres, while mysql doesn't allow duplicate constraint name in schema, 

we were not scanning name from pgdump earlier, but now i have added that, so we will need this validation.

Reply:
Thanks Pradip! Good catch.
Let's keep the test.